src/moonlink_connectors/src/rest_ingest/json_converter.rs

```diff
@@ -54,6 +54,34 @@ impl JsonToMoonlinkRowConverter {
                     Err(JsonToMoonlinkRowError::TypeMismatch(field.name().clone()))
                 }
             }
+            DataType::Boolean => {
+                if let Some(b) = value.as_bool() {
+                    Ok(RowValue::Bool(b))
+                } else {
+                    Err(JsonToMoonlinkRowError::TypeMismatch(field.name().clone()))
+                }
```

Comment:
Can you add tests for the DataType::Boolean branch to cover both the happy path (valid integer) and the error path (non-integer input)?
This will ensure we verify the transformation works correctly and that the TypeMismatch error branch is covered.